File: 1845-seat-reservation-manager/1845-seat-reservation-manager.cpp
class SeatManager {
public:
    priority_queue<int,vector<int>,greater<int>> minheap;
    SeatManager(int n) {
        for(int i=1;i<=n;i++)
        {
            minheap.push(i);
        }
    }
    
    int reserve() {
        int a=minheap.top();
        minheap.pop();
        return a;
    }
    
    void unreserve(int seatNumber) {
        minheap.push(seatNumber);
        
    }
};

/**
 * Your SeatManager object will be instantiated and called as such:
 * SeatManager* obj = new SeatManager(n);
 * int param_1 = obj->reserve();
 * obj->unreserve(seatNumber);
 */

File: 1845-seat-reservation-manager/1845-seat-reservation-manager.py
class SeatManager:

    def __init__(self, n: int):
        self.unres=[i for i in range(1,n+1)]

    def reserve(self) -> int:
        return heappop(self.unres)

    def unreserve(self, seatNumber: int) -> None:
        heappush(self.unres,seatNumber)
        


# Your SeatManager object will be instantiated and called as such:
# obj = SeatManager(n)
# param_1 = obj.reserve()
# obj.unreserve(seatNumber)